code/game/gamemodes/objectives/_objective.dm

```diff
@@ -238,10 +238,14 @@ GLOBAL_LIST(admin_objective_list) //Prefilled admin assignable objective list
 					/area/holodeck,
 					/area/lawoffice,
 				)
+				// If our airlock isn't accessible to these accesses, then we won't allow the item to spawn here
+				var/list/safe_access_list = list(ACCESS_CARGO, ACCESS_MAINT_TUNNELS, ACCESS_MEDICAL, ACCESS_MORGUE, ACCESS_JANITOR, ACCESS_CHAPEL_OFFICE, ACCESS_THEATRE, ACCESS_LAWYER, ACCESS_CONSTRUCTION, ACCESS_MAILSORTING)
```

Comment:
I feel maint airlocks would be too many and hard to find

Reply:
It has to pass the area check and the access check, so it can only pick maint airlocks if they are in the whitelisted areas. I believe all maps have labelled maintenance doors, so the name of it will be pretty clear "Chemistry Maintenance Access in chemistry"